### ISSUE TITLE
Generate debuginfo even when there is no debuginfo in the wasm file

### ISSUE THREAD

**Issue by bjorn3** [CraneStation/wasmtime#439]
For example, wasmtime could use the function index within the wasm file as function name. It could also expose the wasm stack as fake variables, and when possible, it could show the arguments.

It would also be possible (probably behind a flag) to create .wat files and generate appropriate `.debug_line` contents.

These things would significantly ease debugging of wasm programs without debuginfo.

**Comment by tschneidereit** [comment]
Agreed, and this is something we've talked about. Synthesizing at least
rudimentary debug info wouldn't be a huge amount of work, but it's also not
something we'll be able to get to within the next few weeks, probably
months :(

**Comment by fitzgen** [comment]
Wasmtime's recent guest debugging supports stepping through the WAT.
